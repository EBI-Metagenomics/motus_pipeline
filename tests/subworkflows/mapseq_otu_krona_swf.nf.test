nextflow_process {

    name "Test SubWorkflow mapseq-otu-krona"
    script "subworkflows/mapseq_otu_krona_swf.nf"
    process "MAPSEQ_OTU_KRONA"

    test("mapseq-otu-krona") {

        when {
            params {
                seq = "${baseDir}/tests/subworkflows/fixtures/mapseq/test.fasta"
                db = "${baseDir}/tests/modules/fixtures/mapseq/silva-SSU"
                fasta = "SSU.fasta"
                tax = "slv_ssu_filtered2.txt"
                otu = "test-otu"
            }
            process {
                """
                input[0] = channel.fromPath("${params.seq}", checkIfExists: true)
                input[1] = channel.fromPath("${params.db}", checkIfExists: true)
                input[2] = channel.value(params.otu)
                input[3] = channel.value(params.fasta)
                input[4] = channel.value(params.tax)
                input[5] = channel.value("test")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }
}