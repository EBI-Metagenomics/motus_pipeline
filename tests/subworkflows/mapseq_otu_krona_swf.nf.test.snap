{
    "mapseq-otu-krona": {
        "content": [
            {
                "mapseq": [
                    "test.mseq:md5,10f996abe54bd0d5b5429223c9320c47"
                ],
                "biom": [
                    "test.txt:md5,cb95d683b6a7c64da1ef16d1a194c83a"
                ]
            }
        ],
        "timestamp": "2023-03-22T16:33:05+0000"
    }
}